src/StringCodeUtilities.hpp

```diff
@@ -0,0 +1,100 @@
+#ifndef STRING_CODE_UTILITIES
+#define STRING_CODE_UTILITIES
+
+#include <string>
+#include <sstream>
+#include <vector>
+#include <limits>
+
+using namespace std;
+
+/*
+--- String code system of StringCodeComponents ---
```

Comment:
I must be honest: I do not like this way of serialize classes...
However, it can be changed in the future, so probably for now it is ok. But just from the notes you can see that it can be the final solution.
BUT I think that you should explain it better than in here. I have problem in understanding from this explanation. Especially because it seems to cover quite a general collection of situations (it is very abstract) and it miss concrete, solid examples with the FFNN. Perhaps try to add concrete FFNN examples - in the end examples are the best way to teach/learn (Richard Feynman :P)

Reply:
My dislike and love for the solution kind of weigh off each other :-D 
I will also be honest: If this project was not a computational science project, but anything remotely real-world and customer oriented, I would throw it away immediately^^
On the other hand I find my system actually pretty decent to work with in practice. And the string code structure is purely horizontal and any verticality comes only from multiple codes printed into a file. So if the codes itself are not too long, I think it is a nice use of screen space.

Reply:
Besides that I'm disappointed that you had trouble to follow the explanation :-( Guess I will have to look over it again..